BlueprintUICommonControls/Sources/EditingMenu.swift

```diff
@@ -0,0 +1,496 @@
+            case .system: return false
+            case .custom: return true
+            }
+        }
+
+        var isSystem: Bool {
+            switch self {
+            case .system: return true
+            case .custom: return false
+            }
+        }
+
+        var asMenuItem: UIMenuItem? {
+            switch self {
+            case .system:
+                return nil
+
+            case .custom(let title, let selector, _):
+                return UIMenuItem(title: title, action: selector)
+            }
+        }
+    }
+
+    fileprivate static func resolved(with items: [EditingMenuItem]) -> [Resolved] {
+
+        var customSelectors: [Selector] = [
+            #selector(EditingMenu.View.customAction0),
+            #selector(EditingMenu.View.customAction1),
+            #selector(EditingMenu.View.customAction2),
+            #selector(EditingMenu.View.customAction3),
+            #selector(EditingMenu.View.customAction4),
+            #selector(EditingMenu.View.customAction5),
+            #selector(EditingMenu.View.customAction6),
+            #selector(EditingMenu.View.customAction7),
+            #selector(EditingMenu.View.customAction8),
+            #selector(EditingMenu.View.customAction9),
+        ]
+
+        return items.map { item in
+
+            switch item.kind {
+            case .custom(let title):
+                precondition(
+                    customSelectors.isEmpty == false,
+                    """
+                    More than 10 custom actions were provided which is not supported.
+                    More custom action selectors need to be added.
+                    """
+                )
+
+                return .custom(title, customSelectors.removeFirst(), item.onSelect)
+
+            case .system(let kind):
+                return .system(kind, item.onSelect)
+            }
+        }
+    }
+}
+
+extension EditingMenuItem.Kind.System {
+
+    fileprivate var selector: Selector {
+
+        let actions = UIResponderStandardEditActions.self
+
+        switch self {
+        case .cut:
+            return #selector(actions.cut)
+        case .copy:
+            return #selector(actions.copy)
+        case .paste:
+            return #selector(actions.paste)
+        case .delete:
+            return #selector(actions.delete)
+        case .select:
+            return #selector(actions.select)
+        case .selectAll:
+            return #selector(actions.selectAll)
+        case .toggleBoldface:
+            return #selector(actions.toggleBoldface)
+        case .toggleItalics:
+            return #selector(actions.toggleItalics)
+        case .toggleUnderline:
+            return #selector(actions.toggleUnderline)
+        case .increaseSize:
+            return #selector(actions.increaseSize)
+        case .decreaseSize:
+            return #selector(actions.decreaseSize)
+
+        case .printContent:
+            guard #available(iOS 15.0, *) else { fatalError() }
```

Comment:
Nit: Do we need these available checks if the case itself is `@available` checked?

Reply:
We do, weirdly enough